sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuild.java

```diff
@@ -121,6 +134,10 @@ public boolean isDebugLogs() {
 
   public ScannerForMSBuild setScannerVersion(String s) {
     checkArgument(!isEmpty(s), "version must be set");
+    if (s.equals(LATEST_RELEASE)) {
+      s = gitHub.getLatestScannerReleaseVersion().orElse(s);
```

Comment:
It is unclear to me why we are using `orElse(s)`, since it seems destined to fail later on.

Assuming that for any reason, `getLatestScannerReleaseVersion` doesn't manage to retrieve the version, I think we should not go any further, and fail here explicitly, since there is no possibility of the following code to work correctly (to my understanding), instead of failing for a potentially less clear reason afterwards.

Reply:
Indeed, my intention was to let it run with the default version but it doesn't work as expected since I should have returned `null` instead.  And thinking more about this, it would not be great since it would not work as configured.

I will remove the Optional and let the exception bubble up.